parallel-install/pkg/config/kubeconfigmanager.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeConfigManager handles kubeconfig path and kubeconfig content.
+type kubeConfigManager struct {
+	path    string
+	content string
+}
+
+// NewKubeConfigManager creates a new instance of KubeConfigManager.
+func NewKubeConfigManager(path, content *string) (*kubeConfigManager, error) {
+	pathExists := exists(path)
+	contentExists := exists(content)
+	var resolvedPath string
+	var resolvedContent string
+
+	if pathExists && contentExists {
+		resolvedPath = *path
+	} else if pathExists {
+		resolvedPath = *path
+	} else if contentExists {
+		resolvedContent = *content
+	} else {
+		return nil, errors.New("either kubeconfig or kubeconfigcontent property has to be set")
+	}
+
+	return &kubeConfigManager{
+		path:    resolvedPath,
+		content: resolvedContent,
+	}, nil
+}
+
+// Path returns a path to the kubeconfig file.
+func (k *kubeConfigManager) Path() (string, error) {
+	return k.resolvePath()
+}
+
+// Config returns a kubeconfig REST Config used by k8s clients.
+func (k *kubeConfigManager) Config() (*rest.Config, error) {
+	if exists(&k.path) {
+		resolvedPath, err := k.resolvePath()
```

Comment:
i think that in this case we should use what we have instead of creating file if path is empty. if client will provide kubeconfig content then there's no reason to create file for him. we only need file in case of helm scenario

Reply:
and i just noticed that creation of file won't be executed in this case

Reply:
shouldn't we use just k.path in this case?

Reply:
To simplify things it can be changed to `k.path`.